src/Chapter1.hs

```diff
@@ -641,7 +650,10 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n
+  | n == mod (abs n) 10 = n
+  | otherwise = firstDigit (div n 10)
```

Comment:
Here you will need to work with `abs` too to avoid hanging 🙂 

You can see that `abs n` is reused several times in this function. This usually means that you can move it into the `where` case and reuse everywhere 👍🏼